components/browser/session/src/main/java/mozilla/components/browser/session/engine/EngineObserver.kt

```diff
@@ -303,6 +305,73 @@ internal class EngineObserver(
         mediaMap.remove(media)
     }
 
+    override fun onMediaActivated(mediaSession: MediaSession) {
+        val store = store ?: return
+        store.dispatch(MediaSessionAction.ActivatedMediaSessionAction(session.id, mediaSession))
```

Comment:
nit: those can be written as `store?.dispatch(..)` - with that you do not need the check above.